text/0711-deprecate-auto-location.md

```diff
@@ -0,0 +1,140 @@
+---
+Stage: Accepted
+Start Date: 2021-01-23
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: https://github.com/emberjs/rfcs/pull/711
+---
+
+<!---
+Directions for above:
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# <RFC title>
+
+## Summary
+
+Deprecate Ember.AutoLocation and related APIs. In next major version,
+make history the default location (instead of auto).
+
+## Motivation
+
+In practice, 'auto' will almost always resolve to the 'history' location.
+
+By removing the 'auto' location and setting 'history' as the default we're removing
+some complexity around router location, with no downside for users, since they
+all get it resolved to history anyway.
+
+## Background
+
+The Ember Router can use different mechanisms to serialize its state.
+The two common ones are 'history' and 'hash'. There is also a special
+location called 'auto' which does feature detection, preferring 'history'.
+
+Basically all browsers will get 'history':
+https://caniuse.com/mdn-api_history_pushstate
+
+The check 'auto' is basically doing this:
+
+```js
+// Boosted from Modernizr: https://github.com/Modernizr/Modernizr/blob/master/feature-detects/history.js
+// The stock browser on Android 2.2 & 2.3, and 4.0.x returns positive on history support
+// Unfortunately support is really buggy and there is no clean way to detect
+// these bugs, so we fall back to a user agent sniff :(
+
+// We only want Android 2 and 4.0, stock browser, and not Chrome which identifies
+// itself as 'Mobile Safari' as well, nor Windows Phone.
+
+if (
+    (userAgent.indexOf('Android 2.') !== -1 || userAgent.indexOf('Android 4.0') !== -1) &&
+    userAgent.indexOf('Mobile Safari') !== -1 &&
+    userAgent.indexOf('Chrome') === -1 &&
+    userAgent.indexOf('Windows Phone') === -1
+  ) {
+    return false;
+  }
+
+  return Boolean(window.history && 'pushState' in window.history);
+```
+
+## Transition Path
+
+We'll begin with deprecating auto, in next major release we
+set history as the default. Since auto is default today,
+and almost always resolve to history, most users won't need
+to do a thing and they'll still get history location in the
+next major release (but set directly, without auto resolving
+to it).
```

Comment:
I think you will have to call out changes to ember-cli blueprint (in order to avoid the deprecation), and that change will impact all users.